lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,235 @@
+                if (status.EntriesFailed.Count != 0)
+                {
+                    throw new AdlsException($"{status.EntriesFailed.Count} entries did not get transferred correctly");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new AdlsException(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Creates a new directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory cannot be created</exception>
+        public void CreateDirectory(Uri directoryUri)
+        {
+            bool createDirStatus = _adlsClient.CreateDirectory(directoryUri.AbsolutePath);
+            if (!createDirStatus)
+            {
+                throw new AdlsException($"Cannot create directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Deletes a directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory cannot be deleted</exception>
+        public void DeleteDirectory(Uri directoryUri)
+        {
+            bool deleteStatus = Exists(directoryUri) && 
+                _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY &&
+                _adlsClient.DeleteRecursive(directoryUri.AbsolutePath);
+            if (!deleteStatus)
+            {
+                throw new AdlsException($"Cannot delete directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Get the children on the given URI, if that refers to a directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory does not exist</exception>
+        public IEnumerable<Uri> GetChildren(Uri directoryUri)
+        {
+            if (Exists(directoryUri) && _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY)
+            {
+                foreach (var entry in _adlsClient.EnumerateDirectory(directoryUri.AbsolutePath))
+                {
+                    yield return new Uri($"{GetUriPrefix()}{entry.FullName}");
+                }
+            }
+            else
+            {
+                throw new AdlsException($"Cannot find directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Create Uri from a given file path.
+        /// The file path can be full with prefix or relative without prefix.
+        /// If null is passed as the path, ArgumentException will be thrown.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">If specified path is null</exception>
+        public Uri CreateUriForPath(string path)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException("Specified path is null");
```

Comment:
First argument of this exception must be the exact name of the null parameter. Do
```C#
throw new ArgumentNullException(nameof(path), "Specified path is null");
```

Reply:
done